src/child.js

```diff
@@ -20,6 +20,12 @@ class Child {
     if (typeof config.shouldInitImmediately === 'undefined') {
       config.shouldInitImmediately = true;
     }
+    if (typeof config.parse === 'undefined') {
+      config.parse = JSON.parse;
+    }
+    if (typeof config.stringify === 'undefined') {
```

Comment:
We should also check the type of user-provided `parse` and `stringify`. This will prevent the code from breaking because we assume them to be methods only. So, the code might look like the following.

```
const defaultDataStringifier = JSON.stringify;
const defaultDataParser = JSON.parse;

if (config.stringify && Object.prototype.toString.call(config.stringify) === '[object Function]') {
  config.stringify = config.stringify
} else {
  config.stringify = defaultDataStringifier
}

if (config.parse && Object.prototype.toString.call(config.parse) === '[object Function]') {
  config.parse = config.parse
} else {
  config.parse = defaultDataStringifier
}
```

Reply:
I can modify the code to only apply the method if it is a function, but a couple reasons I didn't:

1) The existing code does not do the same type check when defaulting other config values.

```js
// existing code
if (typeof config.heartBeatInterval === 'undefined') {
  config.heartBeatInterval = 500;
}

// more robust check
if (typeof config.heartBeatInterval === 'number') {
  config.heartBeatInterval = 500;
}
```

2) If the user passes bunk, then it will "silently fail" because it will just fall back to the default without notifying the user that its invalid.

I can make the change, I just wanted to call it out.